src/Application/Commands/PatchFunds8Oct.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+namespace MatchBot\Application\Commands;
+
+use Doctrine\ORM\EntityManagerInterface;
+use MatchBot\Application\Matching\Allocator;
+use MatchBot\Application\Matching\MatchFundsRedistributor;
+use MatchBot\Application\Messenger\DonationUpserted;
+use MatchBot\Domain\Campaign;
+use MatchBot\Domain\CampaignFundingRepository;
+use MatchBot\Domain\CampaignRepository;
+use MatchBot\Domain\DoctrineDonationRepository;
+use MatchBot\Domain\Donation;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Messenger\RoutableMessageBus;
+
+/**
+ * @see MatchFundsRedistributor which has some of the core logic in common.
+ */
+#[AsCommand(
+    name: 'matchbot:patch-funds-8-oct',
+    description: 'Fix some incorrect fundings',
+)]
+class PatchFunds8Oct extends Command
+{
+    public function __construct(
+        private Allocator $allocator,
+        private CampaignRepository $campaignRepository,
+        private CampaignFundingRepository $campaignFundingRepository,
+        private DoctrineDonationRepository $donationRepository,
+        private EntityManagerInterface $entityManager,
+        private LoggerInterface $logger,
+        private RoutableMessageBus $bus,
+    ) {
+        parent::__construct();
+    }
+
+    #[\Override]
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $campaigns = $this->getCampaignsAffected();
+        $output->writeln('Campaigns affected: ' . count($campaigns));
+
+        foreach ($campaigns as $campaign) {
+            $campaignId = $campaign->getId() ?? 0;
+            $donations = $this->donationRepository->findWithBigGiveWgmf25Matching($campaign);
+            $output->writeln("Campaign $campaignId - Donations affected: " . count($donations));
+
+            foreach ($donations as $donation) {
+                $this->switchMatchingToCorrectFunding($donation);
+                $output->writeln('Donation ' . ($donation->getUuid()->toString()) . ' matching moved');
+            }
+
+            $this->campaignFundingRepository->zeroBigGiveWgmf25Funding($campaign);
+            $output->writeln("Campaign $campaignId processed + Big Give WGMF25 funding zeroed");
+        }
+
+        return Command::SUCCESS;
+    }
+
+    /**
+     * @return Campaign[]
+     */
+    private function getCampaignsAffected(): array
+    {
+        // Raw pMA export
```

Comment:
may be useful for ease of double checking to add to this comment what query was run in pMA ?

Reply:
Assuming this of from the "better query" from Jira. No need to add the comment to the code, only important that we're clear between us in the short term.